Compiling openvas-scanner branch openvas-scanner-5.1: error cast between incompatible pointer types

<!--
Scanning dependencies of target openvassd
[  6%] Building C object src/CMakeFiles/openvassd.dir/attack.c.o
/home/xxx/src/gvm/openvas-scanner/src/attack.c: In function ‘attack_network’:
/home/xxx/src/gvm/openvas-scanner/src/attack.c:1093:29: error: cast between incompatible function types 
from ‘void (*)(struct attack_start_args *)’ to ‘int (*)(void *)’ [-Werror=cast-function-type]
       pid = create_process ((process_func_t) attack_start, &args);
                         ^
cc1: all warnings being treated as errors
make[2]: *** [src/CMakeFiles/openvassd.dir/build.make:63: src/CMakeFiles/openvassd.dir/attack.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:128: src/CMakeFiles/openvassd.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
-->

### Expected behavior
Just warn or cast in a way not producing warnings or errors.

### Current behavior
Compiler breaks, because _-Werror=cast-function-type_ is set.

### Steps to reproduce
1. Download or checkout sources
2. Solve dependencies
3. Compile

### Environment
**Operating system:** Debian buster/sid

**Installation method / source:** source installation

```
HEAD is now at b9310b2 Merge pull request #212 from jjnicola/fix-no-nvt-category
Already up to date.
Already on 'openvas-scanner-5.1'
Your branch is up to date with 'origin/openvas-scanner-5.1'.
```

It is fixed in 'master', but not backported to 'openvas-scanner-5.1'.

I have an error when I installed scanner-5.1,and this is a bit like it .I don't konw how to solve it ,the error is as follow 👍

attack.c:1093:29: error: cast between incompatible function types from ‘void (*)(struct attack_start_args *)’ to ‘int (*)(void *)’ [-Werror=cast-function-type]
       pid = create_process ((process_func_t) attack_start, &args);

--- a/src/ntp.c
+++ b/src/ntp.c
@@ -391,7 +391,7 @@ __ntp_timestamp_scan (int soc, char *msg)
 static int
 __ntp_timestamp_scan_host (int soc, char *msg, char *host)
 {
-  char timestr[1024];
+  char timestr[64];
   char *tmp;
   time_t t;
   int len;

--- a/src/pluginload.c
+++ b/src/pluginload.c
@@ -250,7 +250,7 @@ cleanup_leftovers (int num_files)
   g_slist_free_full (oids, g_free);
 }
 
-static int
+static void
 plugins_reload_from_dir (void *folder)
 {
   GSList *files = NULL, *f;

--- a/src/processes.h
+++ b/src/processes.h
@@ -28,7 +28,7 @@
 #ifndef _OPENVAS_THREADS_H
 #define _OPENVAS_THREADS_H
 
-typedef int (*process_func_t) (void *);
+typedef void (*process_func_t) (void *);
 pid_t create_process (process_func_t, void *);
 int terminate_process (pid_t);
 

